### DISCUSSION TITLE
404 from backend not picked up by revalidate

### DISCUSSION THREAD

**Discussion by vincentleeuwen** [vercel/next.js#22137] [Help]
Hey guys, hope this is the right place to ask this, otherwise feel free to point me elsewhere!

I'm using static regeneration which generally works great, unless I depublish items in my backend. At this point, my backend returns a 404, but for some reason my page won't revalidate into a 404... (I'm on Next 10.0.6).

My component:

```
export async function getStaticProps({ params: { handle } }) {
  try {
    const { data } = await axios({
      method: 'get',
      url: `${process.env.BACKEND_URL}api/published-product/${encodeURI(
        handle
      )}/`,
      headers: {
        Authorization: `Token ${process.env.BACKEND_AUTH_TOKEN}`,
      },
    });
    const product = { handle, ...data };
    return {
      props: {
        key: `published-product-${handle}`, // https://github.com/vercel/next.js/issues/9992
        product,
      },
      revalidate: 1,
    };
  } catch (error) {
    if (error?.response?.status === 404) {
      return { notFound: true };
    }
    throw error;
  }
}

export async function getStaticPaths() {
  return { paths: [], fallback: true };
}

```

If i redeploy this is solved but i was expecting the revalidation would pick this up? Any idea what i'm doing wrong? 

Thanks,

Vincent

**Comment by lotharvm** [comment]
@vincentleeuwen, did you eventually manage to fix this? I'm running into the same issue at the moment..
